doc/kubernetes/health-check.md

```diff
@@ -0,0 +1,27 @@
+# Why need health-check
+- Current status
+    - We can only check whether the k8s scheduled container is running or not, couldn't exactly know the real supervisor service inside container is healthy or not
+- Problems should be resolved
+    - Health-check should check whether the real supervisor service is healthy or not
+    - If not healthy, there better be different error codes to help debug
+
+# How do we Implement the health-check 
+- Leverage k8s probe tool
+    - K8s has three kinds of probe, [liveness, readiness and startup probes](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#define-startup-probes) which can help us probe state inside container. we can use startup probe for our scenario.
```

Comment:
There are 3 kinds of probes to handle container failure during different stages.  Could you please describe how sonic k8s handle the 3 situations now for sonic container ?

Reply:
I think the startup probe is enough for our scenario.